src/function/table/system/attach_duckdb.cpp

```diff
@@ -0,0 +1,241 @@
+#include <stdint.h>
+#include "duckdb.hpp"
+#include "duckdb/main/query_result.hpp"
+#include "duckdb/parser/parser.hpp"
+#include "duckdb/parser/expression/cast_expression.hpp"
+#include "duckdb/parser/parsed_data/create_table_function_info.hpp"
+#include "duckdb/common/types/date.hpp"
+#include "duckdb/common/types/timestamp.hpp"
+
+#include <cmath>
+
+namespace duckdb {
+
+// this is private in query_result for some reason, I'd rather include it from there but for now I'll do this
```

Comment:
It is private because you're not supposed to access the internals of the iterator - but are supposed to use it through the `begin` and `end` methods on the query result. Either way, copying the code seems like a bad idea.

I don't think you want to use the `QueryResultIterator` to begin with for this functionality as it is extremely slow and mostly provided as a convenience wrapper. See below.